pkg/apis/cr/v1alpha1/types.go

```diff
@@ -131,6 +132,16 @@ type ActionStatus struct {
 	DeferPhase Phase `json:"deferPhase,omitempty"`
 }
 
+// ActionProgress provides information on the progress of an action.
+type ActionProgress struct {
```

Comment:
should we also add the name of the current phase in progress?

Reply:
phase progress is planned for the next iteration. at this level, we just want to communicate the overall progress without bloating the API spec.